### PR TITLE
Jonas Nylund

### PR DESCRIPTION
**Pull request by jonasnylund**
No idea if I am doing this by how it was intended

### CODE REVIEW

**Review by ajlace — COMMENTED** [review]
Report looks good. It needs to be made clearer in  the code what is part of the assignment. 
For example: Which function corresponded to the matrix-vector product was not clear.

Adding the Google Collab link was great, I could however not access the file. Is it perhaps set to private?

As Lab 2 is included in this PR, it will not get merged until Lab 2 is graded.

File: Lab-2/jonasnylund_lab2.ipynb [hunk 1]
```diff
@@ -0,0 +1,764 @@
...
+          "height": 69
+        }
+      },
+      "source": [
+        "n = np.random.randint(2,10);\n",
+        "m = np.random.randint(n,10);\n",
+        "\n",
+        "A = np.random.rand(m,n);\n",
+        "Q,R = qrFactorize(A);\n",
+        "q,r = np.linalg.qr(A);\n",
```
Comment (ajlace) [inline Lab-2/jonasnylund_lab2.ipynb:326]: You do not use these values to prove that your implementation is correct. You need to modify the tests to show correctness of your qr-factorization.

As your implementation does not follow the algorithm given by the lecture notes, it is more important to show that it is correct.

**Review by ajlace — CHANGES_REQUESTED** [review]
Lab 2 looks good. Make the modifications I have requested and it is done.

File: Lab-2/jonasnylund_lab2.ipynb (hunk 1 above)
Comment (jonasnylund) [reply Lab-2/jonasnylund_lab2.ipynb:326]: I don't understand what you want me to change, as all I can see is your comment in the middle of a formatted text document. What values don't I use? What modifications do you want? Am I not doing the tests as specified in the problem description?

**Review by jonasnylund — COMMENTED** [review]
Any update on this? Have I missed something? If by "You do not use these values to prove that your implementation is correct" you are referencing the values obtained from numpy, they are not actually used for anything, just old code that I had forgotten to remove. Or is this the issue, that I am not comparing with the solution obtained from numpy?

File: Lab-2/jonasnylund_lab2.ipynb (hunk 1 above)
Comment (ajlace) [reply Lab-2/jonasnylund_lab2.ipynb:326]: Yes, I would like you to include tests comparing your solution to the np-solution. Otherwise it looks good.

**Comment by ajlace** [comment]
Lab 3 looks great. The code is very well organized and the report is easy to follow.

Edit: For your comment above; I answered it in the original thread.

**Comment by ajlace** [comment]
Lab 4: 👌

File: Lab-2/jonasnylund_lab2.ipynb (hunk 1 above)
Comment (jonasnylund) [reply Lab-2/jonasnylund_lab2.ipynb:326]: I have now implemented the comparison with numpy

**Review by ajlace — APPROVED** [review]
Lab 5: Everything looks good and correct. One thing to think about: You use n! in your calculations of the Monte Carlo quadrature instead of basing it on the number of points. While, as you state, this works, it can cause errors in form of arithmetic overflow or float point errors etc and is hence adviced against - especially if this is to be used in bigger calculations.

**Pull request merged**